ci-operator/step-registry/upi/install/powervs/cluster/upi-install-powervs-cluster-commands.sh

```diff
@@ -0,0 +1,411 @@
+  echo "OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE is an empty string"
+  exit 1
+fi
+
+NO_OF_RETRY=${NO_OF_RETRY:-"5"}
+
+function retry {
+  cmd=$1
+  for i in $(seq 1 "$NO_OF_RETRY"); do
+    echo "Attempt: $i/$NO_OF_RETRY"
+    ret_code=0
+    $cmd || ret_code=$?
+    if [ $ret_code = 0 ]; then
+      break
+    elif [ "$i" == "$NO_OF_RETRY" ]; then
+      error "All retry attempts failed! Please try running the script again after some time" $ret_code
+    else
+      sleep 30
+    fi
+  done
+}
+
+function setup_upi_workspace(){
+  # Before the workspace is created, download the automation code
+  mkdir -p "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir
+  cd "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir || true
+  curl -sL https://raw.githubusercontent.com/ocp-power-automation/openshift-install-power/"${UPI_AUTOMATION_VERSION}"/openshift-install-powervs -o ./openshift-install-powervs
+  chmod +x ./openshift-install-powervs
+  ./openshift-install-powervs setup
+}
+
+function create_upi_tf_varfile(){
+  export PRIVATE_KEY_FILE="${CLUSTER_PROFILE_DIR}"/ssh-privatekey
+  export PUBLIC_KEY_FILE="${CLUSTER_PROFILE_DIR}"/ssh-publickey
+  export CLUSTER_DOMAIN="${BASE_DOMAIN}"
+  export IBMCLOUD_CIS_CRN="${IBMCLOUD_CIS_CRN}"
+
+  if [ -z "${PUBLIC_KEY_FILE}" ]
+  then
+    echo "ERROR: PUBLIC KEY FILE is not set"
+    return
+  fi
+  if [ -z "${PRIVATE_KEY_FILE}" ]
+  then
+    echo "ERROR: PRIVATE KEY FILE is not set"
+    return
+  fi
+
+  cd "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/ || true
+  cp "${PUBLIC_KEY_FILE}" "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/id_rsa.pub
+  cp "${PRIVATE_KEY_FILE}" "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/id_rsa
+  chmod 0600 "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/id_rsa
+  PULL_SECRET=$(<"${CLUSTER_PROFILE_DIR}/pull-secret")
+  echo "${PULL_SECRET}" > "${IBMCLOUD_HOME_FOLDER}"/ocp-install-dir/pull-secret.txt
+
+  cat <<EOF >${IBMCLOUD_HOME_FOLDER}/ocp-install-dir/var-mac-upi.tfvars
+ibmcloud_region     = "${POWERVS_REGION}"
+ibmcloud_zone       = "${POWERVS_ZONE}"
+service_instance_id = "${POWERVS_SERVICE_INSTANCE_ID}"
+rhel_image_name     = "CentOS-Stream-8"
+rhcos_image_name    = "${RHCOS_IMAGE_NAME}"
+system_type         = "e980"
+cluster_domain      = "${CLUSTER_DOMAIN}"
+cluster_id_prefix   = "rdr-mac-ci"
+bastion   = { memory = "16", processors = "1", "count" = 1 }
+bootstrap = { memory = "16", processors = "0.5", "count" = 1 }
+master    = { memory = "16", processors = "0.5", "count" = 3 }
+worker    = { memory = "16", processors = "0.5", "count" = 2 }
+openshift_install_tarball = "https://mirror.openshift.com/pub/openshift-v4/multi/clients/ocp-dev-preview/${OCP_VERSION}/ppc64le/openshift-install-linux.tar.gz"
+openshift_client_tarball  = "https://mirror.openshift.com/pub/openshift-v4/multi/clients/ocp-dev-preview/${OCP_VERSION}/ppc64le/openshift-client-linux.tar.gz"
+release_image_override    = "quay.io/openshift-release-dev/ocp-release:${OCP_VERSION}-multi"
+ibm_cloud_cis_crn = "${IBMCLOUD_CIS_CRN}"
```

Comment:
Does this use the VPC load balancers? if not, let me know.

```
use_ibm_cloud_services     = true
ibm_cloud_vpc_name         = "rdrxxxx"
ibm_cloud_vpc_subnet_name  = "sn01"
ibm_cloud_resource_group = "xyz-resource-group"
iaas_vpc_region           = "us-east"
ibm_cloud_cis_crn         = "crn:v1:bluemix:public:internet-svcs:global:a/XYZ::"
ibm_cloud_tgw             = "rdr-mac-XYZ-tg"
```

Reply:
Right now no, I will have to update it to use above vars

+ this would also mean we need to add additional logic to create vpc, subnet, tgw as part of the script presteps right? @prb112

Reply:
or at least move the existing logic from vpc workflow to upi workflow presteps? is my understanding correct?

Reply:
you are right - we'd have to put in the UPI workflow presteps. and your understanding is correct. If you need my notes on how to accomplish this, I can share it with you directly.